src/scripts/InstanceScripts/Instance_PitOfSaron.cpp

```diff
@@ -94,7 +94,573 @@ class InstancePitOfSaronScript : public MoonInstanceScript
+        }
+        else
+        {
+            mToxicWaste = AddSpell(SPELL_TOXIC_WASTE, Target_Current, 0, 0, 0);
+            mPoisonNova = AddSpell(SPELL_POISON_NOVA, Target_Self, 0, 5000, 0);
+        }
+
+        // Timers
+        mMightyKickTimer = INVALIDATE_TIMER;
+        mPursueTimer = INVALIDATE_TIMER;
+        mPoisonNovaTimer = INVALIDATE_TIMER;
+        mExplosionBarageTimer = INVALIDATE_TIMER;
+        mToxicWasteTimer = INVALIDATE_TIMER;
+        mShadowBoltTimer = INVALIDATE_TIMER;
+        mExplosionBarageEndTimer = INVALIDATE_TIMER;
+        mSpecialAttackTimer = INVALIDATE_TIMER;
+
+        // Emotes
+        // Krick
+        mKrickAI = dynamic_cast<MoonScriptCreatureAI*>(mInstance->GetCreatureBySqlId(CN_KRICK));
+
+        if (!mKrickAI)
+            mKrickAI = SpawnCreature(CN_KRICK, false);
+
+        mKrickAI->AddEmote(Event_OnCombatStart, 8767);
+        mKrickAI->AddEmote(Event_OnTargetDied, 8768);
+        mKrickAI->AddEmote(Event_OnTargetDied, 8769);
+
+        // Ick Spell Announcements
+        mPursue->AddAnnouncement("Ick is chasing you!");
+        mPoisonNova->AddAnnouncement("Ick begins to unleash a toxic poison cloud!");
+    }
+
+    void OnCombatStart(Unit* pTarget)
+    {
+        Phase = BATTLE;
+
+        // Setip Timers
+        mMightyKickTimer = AddTimer(20000);
+        mSpecialAttackTimer = AddTimer(35000);
+        mToxicWasteTimer = AddTimer(5000);
+        mShadowBoltTimer = AddTimer(15000);
+
+        if (mInstance)
+            mInstance->SetInstanceData(Data_EncounterState, _unit->GetEntry(), State_InProgress);
+
+        ParentClass::OnCombatStart(pTarget);
+    }
+
+    void OnCombatStop(Unit* pTarget)
+    {
+        if (mInstance)
+            mInstance->SetInstanceData(Data_EncounterState, _unit->GetEntry(), State_Performed);
+
+        Phase = OUTRO;
+
+        ParentClass::OnCombatStop(pTarget);
+    }
+
+    void OnTargetDied(Unit* pTarget)
+    {
+        MoonScriptBossAI::OnTargetDied(pTarget);
+    }
+
+    void OnDied(Unit* mKiller)
+    {
+        RemoveAIUpdateEvent();
+    }
+
+    void AIUpdate()
+    {
+        if (Phase == BATTLE)
+            if (!_unit->IsCasting())
+                CastSpells();
+
+        ParentClass::AIUpdate();
+    }
+
+    void CastSpells()
+    {
+        // Mighty Kick
+        if (IsTimerFinished(mMightyKickTimer))
+        {
+            CastSpell(mMightyKick);
+            ResetTimer(mMightyKickTimer, 25000);
+        }
+
+        // Toxic Waste
+        if (IsTimerFinished(mToxicWasteTimer))
+        {
+            CastSpell(mToxicWaste);
+            ResetTimer(mToxicWasteTimer, 5000);
+        }
+
+        // Shadow Bolt
+        if (IsTimerFinished(mShadowBoltTimer))
+        {
+            CastSpell(mShadowBolt);
+            ResetTimer(mShadowBoltTimer, 15000);
+        }
+
+        // Special Attack
+        if (IsTimerFinished(mSpecialAttackTimer))
+        {
+            switch (RandomUInt(2))
+            {
+            case 0:
```

Comment:
A case label should be indented from its switch statement.
http://www.ascemu.org/wiki/index.php?title=Coding_standards#indent